test/units/cli/test_galaxy.py

```diff
@@ -444,3 +450,123 @@ def test_template_ignore_similar_folder(self):
+    assert metadata.get('homepage', '') == 'http://example.com'
+    assert metadata.get('issues', '') == 'http://example.com/issue/tracker'
+
+    for d in ['docs', 'plugins', 'roles']:
+        assert os.path.isdir(os.path.join(collection_skeleton, d)), \
+            "Expected collection subdirectory {0} doesn't exist".format(d)
+
+
+@pytest.mark.parametrize('collection_skeleton', [
+    ('ansible_test.delete_me_skeleton', os.path.join(os.path.split(__file__)[0], 'test_data', 'collection_skeleton')),
+], indirect=True)
+def test_collection_skeleton(collection_skeleton):
+    meta_path = os.path.join(collection_skeleton, 'galaxy.yml')
+    assert os.path.exists(meta_path)
+
+    with open(meta_path, 'r') as galaxy_meta:
+        metadata = yaml.safe_load(galaxy_meta)
+
+    for item in ['namespace', 'name', 'version', 'authors']:
+        assert item in metadata, 'unable to find {0}'.format(item)
+
+    assert metadata.get('namespace', '') == 'ansible_test'
+    assert metadata.get('name', '') == 'delete_me_skeleton'
+    assert metadata.get('authors', []) == ['Ansible Cow <acow@bovineuniversity.edu>',
+                                           'Tu Cow <tucow@bovineuniversity.edu>']
+    assert metadata.get('version', '') == '0.1.0'
+
+    assert os.path.exists(os.path.join(collection_skeleton, 'README.md'))
+
+    # Test empty directories exist and are empty
+    for empty_dir in ['plugins/action', 'plugins/filter', 'plugins/inventory', 'plugins/lookup',
+                      'plugins/module_utils', 'plugins/modules']:
+
+        assert os.path.isdir(os.path.join(collection_skeleton, empty_dir))
+        assert os.listdir(os.path.join(collection_skeleton, empty_dir)) == []
+
+    # Test files that don't end with .j2 were not templated
+    doc_file = os.path.join(collection_skeleton, 'docs', 'My Collection.md')
+    assert os.path.exists(doc_file)
+    with open(doc_file, 'r') as f:
+        doc_contents = f.read()
+    assert doc_contents.strip() == 'Welcome to my test collection doc for {{ namespace }}.'
+
+    # Test files that end with .j2 but are in the templates directory were not templated
+    for template_dir in ['playbooks/templates', 'playbooks/templates/subfolder',
+                         'roles/common/templates', 'roles/common/templates/subfolder']:
+        test_conf_j2 = os.path.join(collection_skeleton, template_dir, 'test.conf.j2')
+        assert os.path.exists(test_conf_j2)
+
+        with open(test_conf_j2, 'r') as f:
+            contents = f.read()
+        expected_contents = '[defaults]\ntest_key = {{ test_variable }}'
+
+        assert expected_contents == contents.strip()
+
+
+def test_invalid_skeleton_path():
+    expected = "- the skeleton path '/fake/path' does not exist, cannot init collection"
+    with pytest.raises(AnsibleError, match=expected):
+        gc = GalaxyCLI(args=['ansible-galaxy', 'collection', 'init', 'my.collection', '--collection-skeleton',
```

Comment:
If the exception is only expected in the `run()` method, move this instruction outside of the context manager scope.